### ISSUE TITLE
Table: Global filter match mode NOT_CONTAINS does not work

### ISSUE THREAD

**Issue by sotjdisc** [primefaces/primeng#15009]
### Describe the bug

Global filtering a table with match mode 'notContains' does not work

### Environment

Reproducible at stackblitz

### Reproducer

https://stackblitz.com/edit/cx2ynl?file=src%2Fapp%2Fdemo%2Ftable-filter-menu-demo.html

### Angular version

17.2.2

### PrimeNG version

17.10.0

### Build / Runtime

Angular CLI App

### Language

TypeScript

### Node version (for AoT issues node --version)

Stackblitz node version

### Browser(s)

Chrome 122.0.6261.112

### Steps to reproduce the behavior

Type on the global search box 'James'.

### Expected behavior

All rows containing 'James' must be filtered out.

**Comment by sotjdisc** [comment]
Still persists

**Comment by juri-sinitson** [comment]
Yes, it definitely persists, see [here](https://github.com/primefaces/primeng/blob/6e18b0f740ac10168f99f5b7293e5596f1684615/packages/primeng/src/table/table.ts#L5522) I've tested [this](https://github.com/primefaces/primeng/tree/v18-prod) branch, it should be the most recent one.

For me it looks like concatenating the match for each field with the and-operator would solve the problem.

Maybe it's reasonable to put the filtering stuff to the TableService and introduce unit testing for it beginning with the filtering. I think the 'notContains' used globally would be a good candidate for first tests.

I think it's doable for me to fix that bug during this week including the unit tests for global notContains operator.

Does it make sense for you @cetincakiroglu and @cagataycivici?

**Comment by juri-sinitson** [comment]
[This](https://github.com/primefaces/primeng/pull/16853) PR fixes the issue
